pkg/controllers/chart/chart.go

```diff
@@ -55,6 +55,7 @@ var (
 	deletePolicy         = metav1.DeletePropagationForeground
 	DefaultJobImage      = "rancher/klipper-helm:v0.8.0-build20230510"
 	DefaultFailurePolicy = FailurePolicyReinstall
+	defaultBackOffLimit  = int32(1000)
```

Comment:
Does backoff 1000 really make sense? even suspect k8s really supports 1000.

Reply:
that was the default prior to this change.

Reply:
The back off limit can be any positive integer: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.25/#jobspec-v1-batch

Reply:
do we have a preference to make the default smaller?

Reply:
No, it should remain the same, since there doesn't appear to be any good reason to change the default as part of this enhancement.

Reply:
As said `Failed Pods associated with the Job are recreated by the Job controller with an exponential back-off delay (10s, 20s, 40s …)`,  can't imagine how many years ? will it be after 1000 times of exponential back-off.

We may leave it unchanged now. 

But it is still a question to me: if HelmChart plans to let to job run `forever`, the `exponential  BackOffLimit` looks not a good option, after a certain times of failure, the next try will happen after days/weeks/....  In such cases, even try to run the job each hour will be a better option.

Reply:
If you'd like to propose changing it, please open an issue and related PR. That is not within the scope of what we are discussing here in this enhancement.